app/scripts/app-init.js

```diff
@@ -111,14 +99,13 @@ function importAllScripts() {
   }
 }
 
-// eslint-disable-next-line no-undef
-self.addEventListener('install', importAllScripts);
+importAllScripts();
 
 /*
  * Message event listener below loads script if they are no longer available.
  * chrome below needs to be replaced by cross-browser object,
  * but there is issue in importing webextension-polyfill into service worker.
```

Comment:
The part of the polyfill that gives you this particular functionality should be under 20 lines. It'd be wise to copy-paste it here.

Reply:
We are using this polyfill library https://github.com/MetaMask/metamask-extension/blob/develop/package.json#L380

We need to figure out something different for service worker to avoid pasting whole thing here.

Reply:
Yes, and here we only need a function to add a message listener, which seems to be a matter of getting the `runtime` reference depending on the environment.

we don't seem to be using anything else here and rightfully so.

Reply:
I've added a suggestion to the comment above. Feel free to remove the return statement or keep it - whichever you find less surprising.